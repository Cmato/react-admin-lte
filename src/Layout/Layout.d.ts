import * as React from 'react';

export interface LayoutProps {
  boxed?: boolean;
  fixed?: boolean;
  miniSidebar?: boolean;
  sidebarCollapsed?: boolean;
  topNav?: boolean;
}

declare const Layout: React.ComponentType<LayoutProps>;

export default Layout;
