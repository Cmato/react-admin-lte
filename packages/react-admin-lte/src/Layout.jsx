import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  boxed: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
  skin: PropTypes.oneOf([
    'blue', 'black', 'purple', 'green', 'red', 'yellow', 'blue-light', 'black-light',
    'purple-light', 'green-light', 'red-light', 'yellow-light',
  ]).isRequired,
};

const defaultProps = {
  boxed: false,
};

const childContextTypes = {
  $adminlte_layout: PropTypes.shape({
    toggleMainSidebar: PropTypes.func,
  }),
};

const contextTypes = {
  document: PropTypes.object,
};

class Layout extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      document: context.document ? context.document : document,
    };
  }

  componentDidMount() {
    this.state.document.body.classList.toggle('sidebar-mini', true);
    this.state.document.body.classList.toggle(`skin-${this.props.skin}`, true);

    if (this.props.boxed) {
      this.state.document.body.classList.toggle('layout-boxed', true);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.skin !== nextProps.skin) {
      this.state.document.body.classList.toggle(`skin-${nextProps.skin}`, true);
      this.state.document.body.classList.toggle(`skin-${this.props.skin}`, false);
    }

    this.state.document.body.classList.toggle('layout-boxed', nextProps.boxed);
  }

  componentWillUnmount() {
    this.state.document.body.classList.toggle('sidebar-mini', false);
    this.state.document.body.classList.toggle(`skin-${this.props.skin}`, false);
    this.state.document.body.classList.toggle('layout-boxed', false);
    this.state.document.body.classList.toggle('sidebar-collapse', false);
  }

  getChildContext() {
    return {
      $adminlte_layout: {
        toggleMainSidebar: () => {
          this.state.document.body.classList.toggle('sidebar-collapse');
        },
        setMainSidebarCollapsed: (val) => {
          this.state.document.body.classList.toggle('sidebar-collapse', val);
        },
      },
    };
  }

  render() {
    const classes = {
      wrapper: true,
    };

    return (
      <div className={classNames(this.props.className, classes)}>
        {this.props.children}
      </div>
    );
  }
}

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;
Layout.childContextTypes = childContextTypes;
Layout.contextTypes = contextTypes;

export default Layout;
