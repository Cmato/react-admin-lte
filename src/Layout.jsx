import React from 'react';
import classNames from 'classnames';

const propTypes = {
  boxed: React.PropTypes.bool,
  children: React.PropTypes.node,
  className: React.PropTypes.string,
  skin: React.PropTypes.oneOf([
    'blue', 'black', 'purple', 'green', 'red', 'yellow', 'blue-light', 'black-light',
    'purple-light', 'green-light', 'red-light', 'yellow-light',
  ]).isRequired,
};

const defaultProps = {
  boxed: false,
};

const childContextTypes = {
  $adminlte_layout: React.PropTypes.shape({
    toggleMainSidebar: React.PropTypes.func,
  }),
};

class Layout extends React.Component {
  componentDidMount() {
    document.body.classList.toggle('sidebar-mini', true);
    document.body.classList.toggle(`skin-${this.props.skin}`, true);

    if (this.props.boxed) {
      document.body.classList.toggle('layout-boxed', true);
    }
  }

  componentWillReceiveProps(nextProps) {
    document.body.classList.toggle(`skin-${nextProps.skin}`, true);
    document.body.classList.toggle(`skin-${this.props.skin}`, true);
    document.body.classList.toggle('layout-boxed', this.props.boxed);
  }

  componentWillUnmount() {
    document.body.classList.toggle('sidebar-mini', false);
    document.body.classList.toggle(`skin-${this.props.skin}`, false);
    document.body.classList.toggle('layout-boxed', false);
    document.body.classList.toggle('sidebar-collapse', false);
  }

  getChildContext() {
    return {
      $adminlte_layout: {
        toggleMainSidebar: () => {
          document.body.classList.toggle('sidebar-collapse');
        },
        setMainSidebarCollapsed: (val) => {
          document.body.classList.toggle('sidebar-collapse', val);
        },
      },
    };
  }

  render() {
    const classes = {
      wrapper: true,
    };

    return (
      <div className={classNames(this.props.className, classes)}>
        {this.props.children}
      </div>
    );
  }
}

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;
Layout.childContextTypes = childContextTypes;

export default Layout;
