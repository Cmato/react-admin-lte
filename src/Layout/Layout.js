import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import warning from 'warning';
import uncontrollable from 'uncontrollable';

class Layout extends Component {
  constructor(props) {
    super(props);
    this.handleToggle = this.handleToggle.bind(this);
  }

  getChildContext() {
    return {
      $adminlte_layout: {
        onToggle: this.handleToggle
      }
    };
  }

  handleToggle(e) {
    this.props.onToggle(!this.props.sidebarCollapsed, e);
  };

  render() {
    const {
      boxed,
      children,
      className: classNameProp,
      fixed,
      miniSidebar,
      onSidebarCollapsedToggle: _,
      sidebarCollapsed,
      topNav,
      ...other
    } = this.props;

    warning(
      !(boxed === true && fixed === true),
      'The boxed and fixed properties should not be supplied at the same time.'
    );

    const className = classNames({
      fixed: fixed,
      'layout-boxed': boxed,
      'layout-top-nav': topNav,
      'sidebar-collapse': sidebarCollapsed,
      'sidebar-mini': miniSidebar
    }, classNameProp);

    return (
      <div className={className} {...other}>
        {children}
      </div>
    );
  }
}

Layout.propTypes = {
  /**
   * Layout will only be stretched to 1250px.
   */
  boxed: mutuallyExclusiveTrueProps('boxed', 'fixed'),
  /**
   * The content of the layout.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * It `true`, collapsed the sidebar by default.
   */
  defaultSidebarCollapsed: PropTypes.bool,
  /**
   * Displays fixed header and sidebar.
   */
  fixed: mutuallyExclusiveTrueProps('boxed', 'fixed'),
  /**
   * When collapsed, sidebar will show icons only.
   */
  miniSidebar: PropTypes.bool,
  /**
   * Callback fired when the sidebar is toggled.
   *
   * @param {object} event The event source of the callback
   * @param {boolean} collapsed The 'collapsed` state of the sidebar
   */
  onToggle: PropTypes.func,
  /**
   * If `true`, collapses the sidebar. Setting this prop enables control of
   * the Layout.
   */
  sidebarCollapsed: PropTypes.bool,
  /**
   * Removes sidebar and puts navbar at top.
   */
  topNav: PropTypes.bool
};

Layout.childContextTypes = {
  $adminlte_layout: PropTypes.shape({
    onToggle: PropTypes.func
  })
};

export default uncontrollable(Layout, {
  sidebarCollapsed: 'onToggle'
});

