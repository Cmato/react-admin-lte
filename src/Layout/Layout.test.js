import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import Layout from './Layout';

it('renders empty', () => {
  expect(shallow((
    <Layout className="best-layout">
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders fixed', () => {
  expect(shallow((
    <Layout className="best-layout" fixed={true}>
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders boxed', () => {
  expect(shallow((
    <Layout className="best-layout" boxed={true}>
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders top navigation', () => {
  expect(shallow((
    <Layout className="best-layout" topNav={true}>
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders mini sidebar', () => {
  expect(shallow((
    <Layout className="best-layout" miniSidebar={true}>
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders collapsed sidebar', () => {
  expect(shallow((
    <Layout
      className="best-layout"
      sidebarCollapsed={true}
      onToggle={() => { return; }}
    >
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders default collapsed sidebar', () => {
  expect(shallow((
    <Layout
      className="best-layout"
      defaultSidebarCollapsed={true}
    >
      <div />
    </Layout>
  ))).toMatchSnapshot();
});

it('renders changed collapsed sidebar on toggle', () => {
  const sentEvent = {};
  let receivedCollapsed = false;
  let receivedEvent;

  class SidebarToggler extends Component {
    static contextTypes = {
      $adminlte_layout: PropTypes.shape({
        onToggle: PropTypes.func
      })
    };

    componentDidMount() {
      this.context.$adminlte_layout.onToggle(sentEvent);
    }

    render() {
      return null;
    }
  }

  mount(
    <Layout onToggle={(collapsed, e) => {
      receivedCollapsed = collapsed;
      receivedEvent = e;
    }}>
      <SidebarToggler />
    </Layout>
  );

  expect(receivedCollapsed).toEqual(true);
  expect(receivedEvent).toEqual(sentEvent);
});

