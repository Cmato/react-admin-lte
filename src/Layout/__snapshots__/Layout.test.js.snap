// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders boxed 1`] = `
<Layout
  boxed={true}
  className="best-layout"
  onToggle={[Function]}
>
  <div />
</Layout>
`;

exports[`renders collapsed sidebar 1`] = `
<Layout
  className="best-layout"
  onToggle={[Function]}
  sidebarCollapsed={true}
>
  <div />
</Layout>
`;

exports[`renders default collapsed sidebar 1`] = `
<Layout
  className="best-layout"
  onToggle={[Function]}
  sidebarCollapsed={true}
>
  <div />
</Layout>
`;

exports[`renders empty 1`] = `
<Layout
  className="best-layout"
  onToggle={[Function]}
>
  <div />
</Layout>
`;

exports[`renders fixed 1`] = `
<Layout
  className="best-layout"
  fixed={true}
  onToggle={[Function]}
>
  <div />
</Layout>
`;

exports[`renders mini sidebar 1`] = `
<Layout
  className="best-layout"
  miniSidebar={true}
  onToggle={[Function]}
>
  <div />
</Layout>
`;

exports[`renders top navigation 1`] = `
<Layout
  className="best-layout"
  onToggle={[Function]}
  topNav={true}
>
  <div />
</Layout>
`;
